LPC1769 SKR 1.4 Turbo MAX6675 softSPI not working [BUG]

#19088 # Bug Description

I am building MARLIN 2.0.6.1 LPC1769 on SKR 1.4 Turbo.
I want to use the MAX6675 for my single extruder. I used to have an MKS SBASE with a MAX6675 running without any problem.

I defined the MAX6675 corresponding pins on the SKR 1.4 Pins file, then I declared "#define TEMP_SENSOR_0 -2"  in "configuration.h". The firmware compiles OK but, when the board flashes the firmware from the SD card both the SD Card stops being recognized by the pc and the board cannot communicate over USB serial -1 port (the only one that I currently have enabled).
If I define "#define TEMP_SENSOR_0 1" everything works fine again.
I tried defining different unused pins, for example EXP1 port
```cpp
#if !defined(MAX6675_SS_PIN)    // Thermocouple SS Pins
  #define MAX6675_SS_PIN   P1_20
  #define MAX6675_MISO_PIN P1_18
  #define MAX6675_SCK_PIN  P1_22
#endif
```
The result is always the same. As soon as I define "#define TEMP_SENSOR_0 -2", the controller stops working.

[Config.zip](https://github.com/MarlinFirmware/Marlin/files/5234851/Config.zip)

### My Configurations

**Required:** Please include a ZIP file containing your `Configuration.h` and `Configuration_adv.h` files.

### Steps to Reproduce

<!-- Please describe the steps needed to reproduce the issue -->

With an SKR 1.4 Turbo declare #define TEMP_SENSOR_0 -2" in "configuration.h" file

**Expected behavior:** [What you expect to happen]
Working MAX6675 for my extruder as it used to do it in MARLIN 2.0.5.3 in an MKS_SBASE board (LPC1768)

**Actual behavior:** [What actually happens]
The firmware compiles ok but when I flash it from the SD card, the SKR 1.4 Turbo ceases to work.

I have noticed that there are some example configuration.h files that call the "MAX6675_MISO_PIN" like "MAX6675_DO_PIN",
Therefore, I tried this, and now It won't compile. 
The error states: 
"""
//.platformio/packages/toolchain-gccarmnoneeabi/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld.exe: .pio/build/LPC1769/src/src/module/temperature.cpp.o: in function `Temperature::init()':
//Marlin-2.0.6.1/Marlin\src\module/temperature.cpp:1810: undefined reference to `SPIclass<(unsigned char)9, (unsigned char)18, (unsigned char)7>::softSPI'
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\LPC1769\firmware.elf] Error 1
"""

I have just tried with the last bugfix-2.0.x
If I define "MAX6675_DO_PIN"  then:
The error states:
"""
//.platformio/packages/toolchain-gccarmnoneeabi/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld.exe: .pio/build/LPC1769/src/src/module/temperature.cpp.o: in function Temperature::init()': //Marlin-2.0.6.1/Marlin\src\module/temperature.cpp:1810: undefined reference to SPIclass<(unsigned char)9, (unsigned char)18, (unsigned char)7>::softSPI'
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\LPC1769\firmware.elf] Error 1
"""
If I also "#define LPC_SOFTWARE_SPI" Same error.

If I only declare the "#define MAX6675_SS_PIN    P0_26" which should make use of the MOSI1 and SCK1 pins of the LPC1769. It does compile, but the communication with the MAX6675 still wont work and MAXTEMP triggers on power on.
The MAX6675 module was working until I disassembled the electronics this morning. I will test it again using the MKS SBASE that I removed and confirm that this is a software issue.
However, MAX6675 through SoftSPI should also work and it is currently not compiling.

--- a/Marlin/src/module/temperature.cpp
+++ b/Marlin/src/module/temperature.cpp
@@ -1563,6 +1563,7 @@ void Temperature::updateTemperaturesFromRawValues() {
 }
 
 #if MAX6675_SEPARATE_SPI
+  template<uint8_t MisoPin, uint8_t MosiPin, uint8_t SckPin> SoftSPI<MisoPin, MosiPin, SckPin> SPIclass<MisoPin, MosiPin, SckPin>::softSPI;
   SPIclass<MAX6675_DO_PIN, MOSI_PIN, MAX6675_SCK_PIN> max6675_spi;
 #endif
 

